CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+
+The structure of a project (where the f*** is that file?) might be more important than it's architecture. In this case, there is a three-way split: *backend* and *frontend* as before, but we also have de *e2e* tests to worry about.
+
+But some things also happen at the root:
+
+- **docker-compose.yml**: Defines how the different parts of the project launch and interact with each other.
+- **.travis.yml**: Defines how the different pars of the project are to be tested.
+
+However, we will understand this further in the following sections.
+
+## Backend structure
+
+- **backend/Dockerfile**: Defines how the Docker image for the backend will be built and run
+- **backend/project.clj**: Defines the project dependencies and entry point
+- **backend/test/card_game/**: Contains all the unit tests for the code. Remember that a unit test must be done *before* the actual code.
+- **backend/src/card_game/**: Contains the code that will run in production, the actual logic.
+- **backend/src/card_game/persistence.clj**: This is the only file to interact directly with the database. All database calls *must* pass through here.
+- **backend/src/card_game/api_handler.clj**: This is the only file to interact directly with frontend. Any actions to take as a response to a frontend call *must* be defined here.
+
+## Frontend structure
+
+- **frontend/Dockerfile** and **frontend/entry.sh**: Defines the Docker image and its entrypoint.
+- **frontend/package.json** and **frontend/yarn.lock**: Define the project dependencies (`yarn.lock` is autogenerated by `yarn`)
+- **frontend/test/unit/**: Contains all the unit tests for the code. Remember that a unit test must be done *before* the actual code.
+- **frontend/public/**: The "website", with `index.html` being the entrypoint, but the real magic happens elsewhere...
+- **frontend/src/**: Contains the code that will run in production, with its own structure.
+- **frontend/src/main.js**: The official entrypoint for Vue. Should only be modified if we add a dependency that works through `Vue.use(dependency)`
+- **frontend/src/App.vue**: The main Vue component. Since we're mostly using everything through `router` this mainly hosts the `<router-view/>` and little else.
+- **frontend/src/router.js**: Here the "routes" are defined, which is the way Vue interprets what component or view to serve based on the URL (for example `http://frontend:8080/#/create-game` points to the CreateGame component)
+- **frontend/src/components/** and **frontend/src/views**: The application logic, separated into "views" and "components" following no good criteria as of yet, we will improve on this in the future.
+
+
+# Contributing guidelines
+
+For any change or contribution, this is what is expected of you:
+
+1. Announce within an issue that you want it assigned to you (or open said issue yourself).
+2. Do not take on more than one issue at a time
+3. Design the tests for your issue (unit tests always and in some cases e2e tests may be required)
```

Comment:
I would be very clear with the *one-test-at-a-time* philosophy

1. Make **a single** test that fails
2. Write **as few code as possible** to make it pass the test (and of course the already written tests)
3. Repeat from 1